packages/code-snippet/src/CodeSnippetService.ts

```diff
@@ -70,4 +70,44 @@ export class CodeSnippetService {
       }
     });
   }
+
+  /**
+   *  Duplicates an existing code snippet, which is named
+   *  'Copy of <original-name>'.
+   *
+   * @param codeSnippet: code snippet to be duplicated
+   *
+   * @returns A promise
+   */
+  static duplicateCodeSnippet(
```

Comment:
Given this function is exactly the same as the the code in `MetadataWidget.tsx`, I think it would make more sense to put the util function in `metadata-common` and call it from both places like you called it here in code snippets

Reply:
Done.